prov/gni/include/gnix_ep.h

```diff
@@ -245,6 +245,32 @@ _gnix_fr_alloc(struct gnix_fid_ep *ep)
 	return fr;
 }
 
+static inline struct gnix_fab_req *
+_gnix_fr_alloc_w_cb(struct gnix_fid_ep *ep, void (*cb)(void *))
+{
+	struct dlist_entry *de = NULL;
+	struct gnix_fab_req *fr = NULL;
+	int ret = _gnix_fl_alloc(&de, &ep->fr_freelist);
+
+	while (unlikely(ret == -FI_EAGAIN))
+		ret = _gnix_fl_alloc(&de, &ep->fr_freelist);
```

Comment:
I think this checking - and the point in ```_gnix_fl_alloc``` is coded to return -FI_EAGAIN may be a hold over from some previous way we were dealing with growing the free list and handling of multiple threads using the free list concurrently.  We probably could remove this loop from both fr alloc routines.  I'd rather do that in a separate PR though.